book/src/programmability/witness-pattern.md

```diff
@@ -1,277 +1,117 @@
-    transfer::transfer(coin, to)
+/// Create a new supply for type T with the provided witness.
+public fun create_supply<T: drop>(_w: T): Supply<T> {
+    Supply { value: 0 }
 }
 
-/// Public API - does not require a Witness.
-public fun join<T>(coin: &mut RegulatedCoin<T>, other: RegulatedCoin<T>) {
-    let RegulatedCoin { id, value } = other;
-    coin.value = coin.value + value;
-    id.delete();
+/// Get the `Supply` value.
+public fun supply_value<T>(supply: &Supply<T>): u64 {
+    supply.value
 }
 ```
 
-In the example above, the `mint`, `burn`, and `transfer` functions require a Witness of the `T` type
-to be called. This way, the application module can implement these functions and ensure that only
-the module that owns the `T` can call them. The `join` function, on the other hand, is publicly
-available and does not require a Witness.
+In the example above, which is borrowed from the `balance` module of the
+[Sui Framework](./sui-framework.md), the `Supply` a generic struct that can be constructed only by
+supplying a witness of the type `T`. The witness is taken by value and _discarded_ - hence the `T`
+must have the [drop](./../move-basics/drop-ability.md) ability.
 
-This pattern allows sharing base type and details of the implementation between different
-applications, while still allowing the applications to implement their own logic on top of the base
-type.
-
-## Storable Witnesses + Metadata
-
-It is possible to extend the Witness type by adding the [`store`](./../storage/store-ability.md)
-ability, which would allow the type to be stored in objects. We call this variation of the Witness a
-_Transferable Witness_. However, while this method can be useful in some applications by providing
-the caller a way to perform a _delayed_ authorization or submit a combination of different
-witnesses; it is not recommended to use it in open or sensitive systems, as it may be a security
-risk. The callee function may end up storing the Witness in an object and using it later, which may
-lead to unexpected behaviour.
-
-With that in mind, the _Transferable Witness_ still has some benefits, such as the ability to reuse
-a "metadata" or "configuration" struct and pass it as a Witness. This way, the application can avoid
-unnecessary linking of the metadata with an extra type, and the metadata can _act like a Witness_.
-To illustrate this, let's consider a generic ticketing application: the `Ticket` is a generic type
-which stores metadata, and the application can implement some protected functions for it, such as
-`transfer`.
-
-> In the implementation below we omit certain features (like getter functions and the ability to
-> consume a Ticket) for the sake of clarity and simplicity. The full example is available in the
-> [Move Book](https://github.com/MystenLabs/move-book/tree/main/packages/samples/examples)
-> repository.
+The instantiated `Supply<T>` can then be used to mint new `Balance<T>`'s, where `T` is the type of
```

Comment:
Potential reinforcing point here: It allows the module that defines `T` (the coin marker type) to control when and how supply is created and destroyed.